src/main/webapp/app/exam/participate/exam-navigation-bar/exam-navigation-bar.component.ts

```diff
@@ -45,6 +54,23 @@ export class ExamNavigationBarComponent implements OnInit {
                 this.itemsVisiblePerSide = 0;
             }
         });
+
+        this.exercises
+            .filter((ex) => ex.type === ExerciseType.PROGRAMMING && ex.studentParticipations)
```

Comment:
I can see the following issues:
1) It seems that this code is also executed at the exam start when no student would have been able to make changes (so the REST call would be unnecessary). In an exam with 2000 students and 2 programming exercises, this would lead to 4000 REST calls at the exact same time (i.e. exam start). Those REST calls would all check the default branch on Bitbucket (because repositories are typically not cached on all server nodes). This would immediately kill Bitbucket and would also lead to a very poor performance on the Artemis servers. 

So, while we solve a rare edge case for a few students (who might not even recognize the issue), it would make the user experience of all students worse and could potentially lead to a complete exam failure.

==> Please make sure this code is only executed in case a student triggers a reload. Otherwise, we cannot merge it!

2) When the exam starts, there is no programming submission. I wonder whether the code actually can work in such a case, because in line 70 you use `ExamParticipationService.getSubmissionForExercise(ex)!`. This is dangerous. I vaguely remember that we introduce a "fake programming submission" for this case, but I'm not sure at which point it will be created. You should try to avoid the `!` at all costs if it is possible

3) Minor issue: I don't like short variables such as `ex` or `x`. Please avoid them

Reply:
This is a very valid point @krusche
Now I perform the calls only if the exam page was reloaded, which should address exactly our use-case.